### DISCUSSION TITLE
Update stylex variables on the go

### DISCUSSION THREAD

**Discussion by KylixMedusa** [facebook/stylex#429] [Q&A] [answered]
### Describe the feature request

It's very crucial to update the variables on the go for the kind of use case. We want to use stylex to build a design system, and dynamic theme update is a necessity. Can you suggest some way to achieve that? or maybe pick that up as a feature?

**Comment by KylixMedusa** [comment]
```tsx
export default function styleXDefineVars<Vars: VarsConfig>(
  variables: Vars,
  options: $ReadOnly<{ ...Partial<StyleXOptions>, themeName: string, ... }>,
): [VarsObject<Vars>, { [string]: InjectableStyle }] {
  const { classNamePrefix, themeName } = {
    ...defaultOptions,
    ...options,
  };

  const themeNameHash = classNamePrefix + createHash(themeName);

  const updateVars = (vars: Vars) => { 
    const variablesMap = objMap(variables, (value, key) => {
      // Created hashed variable names with fileName//themeName//key
      const nameHash = classNamePrefix + createHash(`${themeName}.${key}`);
      return { nameHash, value };
    });

    const themeVariablesObject = objMap(variablesMap, ({ nameHash }) => {
      return `var(--${nameHash})`;
    });

    const injectableStyles = constructCssVariablesString(
      variablesMap,
      themeNameHash,
    );

    return [
      {
        ...themeVariablesObject,
        __themeName__: themeNameHash, 
        updateVars
      },
      injectableStyles,
    ]
  }


  return updateVars(variables);
}
```

@nmn we can implement something like this as a fix to this problem i.e. providing a update method that updates the variables on the go

**Comment by necolas** [comment]
> dynamic theme update

Please can you elaborate on the use case with practical examples, and why those examples can't be implemented using the current APIs. Thanks

**Comment by nmn** [comment]
Converting to discussion topic as I'm certain that your use-case is possible with the current APIs. Once you explain what you're trying to do, I'll explain how.

**Comment by KylixMedusa** [comment]
@necolas @nmn 

I'm defining the variables in root using the `defineVars` method. But in our platform we change the color scheme of the platform based on API responses. So during runtime I would need to update the colors into those variables. How can I achieve this?

**Reply by nucleartux** [reply, marked as answer]
I think I asked the same question https://github.com/facebook/stylex/discussions/206
